### DISCUSSION TITLE
Using `ruff` on Windows through `pre-commit` - problem and solution

### DISCUSSION THREAD

**Discussion by jond01** [astral-sh/ruff#4243] [General]
I use `ruff` as a [`pre-commit` hook](https://beta.ruff.rs/docs/usage/#pre-commit) as advised and ran into a problem on Windows (10):
```cmd
pre-commit run ruff --all-files
```
Would output:
```
exit code: 3221225781
```
I first tried to install `ruff` with `pip`, but it didn't help.
What solved the issue for me was installing Rust:
https://www.rust-lang.org/tools/install

I hope this will help users in the future.
Maybe it's even worth mentioning in the installation part if it occurs to many others:
https://beta.ruff.rs/docs/installation/
Currently, there are no special instructions for Windows.

**Comment by MichaReiser** [comment]
Uhh, that's strange. Installing ruff with `pip` should work out of the box on Windows (to my knowledge). 

Can you tell us a little more about your setup. What CPU architecture are you using? Or can you share the log for installing ruff (I've never done this myself but found it on StackOverflow):

```
pip install ruff --log LOG_FILE
```

and then share the content of LOG_FILE with us. If it doesn't work, don't bother with it.

**Comment by charliermarsh** [comment]
Interesting, we do build wheels for Windows so it should work fine without a Rust toolchain. Maybe something specific to the architecture?

**Comment by jond01** [comment]
Thank you both for responding!
To be clear, installing ruff on my machine with **pip** and using it was successful from the start.
The issue occurred only when using ruff as a **pre-commit** hook, which to the best of my knowledge and checks should do the same, but it didn't work smoothly before downloading Rustup (and installing Rust).
The machine's architecture is standard - AMD 64 (`ruff-0.0.265-py3-none-win_amd64.whl` is downloaded with pip).

I've opened this discussion to see if this small issue reoccurs for others, right now it doesn't seem the case 🙂

**Reply by MichaReiser** [reply]
Hmm that's weird. I wonder what's happening. 

I booted into my windows (AMD64), installed pre-commit, and set up the ruff hook. I must admit, I was too lazy to uninstall Rust, but I observed the processes that `pre-commit` spawns and I didn't see any rust process. I'm curious why pre-commit tries to install from source. 

Is my assumption correct that you copied the pre-commit configuration from Ruff's website and aren't using [local paths](https://pre-commit.com/#developing-hooks-interactively)?

**Reply by jond01** [reply]
> Is my assumption correct that you copied the pre-commit configuration from Ruff's website and aren't using [local paths](https://pre-commit.com/#developing-hooks-interactively)?

Yes, correct.

I admit that I tried to reproduce it locally by uninstalling Rust (with rustup), and the issue didn't reoccur. It happened only before installing Rust the first time.
